plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseTestElasticsearchConnectorTest.java

```diff
@@ -969,6 +1116,27 @@ protected void assertTableDoesNotExist(String name)
 
     protected abstract String indexEndpoint(String index, String docId);
 
+    /**
+     * This method overrides the default values used for the data provider
+     * of the test {@link AbstractTestQueries#testLargeIn(int)} by taking
+     * into account that by default Elasticsearch supports only up to `1024`
+     * clauses in query.
+     *
+     * Consult `index.query.bool.max_clause_count` elasticsearch.yml setting
+     * for more details.
+     *
+     * @return the amount of clauses to be used in large queries
+     */
+    @Override
+    protected Object[][] largeInValuesCountData()
+    {
+        return new Object[][] {
+                {200},
+                {500},
+                {1000}
```

Comment:
nit: Maybe use 1024 instead of 1000 since that's the max supported?

Reply:
I've tried with `1024` and the test fails ... I can put `1023` , but I assume that it doesn't make really a difference in the test to test the borders of the default Elasticsearch setting.

Reply:
@losipiuk I'm not sure if I remember correctly but don't connectors indicate to the engine how large in-clauses they can handle? Was it something like domain compaction threshold? Should Elasticsearch connector also be setting it?

Reply:
There is no general mechanism for that for what I see. There is configuration and logic in JDBC based connectors and Hive connector. We could add logic for ES to, but it is not like flipping a flag. A few lines of code would need to be added. Let's keep test refactor and that separate.